### PR TITLE
fix localization error

### PR DESCRIPTION
**Pull request by mjkkirschner**
### Purpose

Don't use localized string to check if workspace is a home workspace.

### Declarations

Check these if you believe they are true

- [ ] The code base is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions), and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

@ramramps 
@QilongTang 

### FYIs

@andydandy74 @kronz @jnealb

### CODE REVIEW

**Comment by mjkkirschner** [comment]
screen shot 2018-03-09 at 4 23 28 pm

File: src/DynamoCore/Graph/Workspaces/WorkspaceInfo.cs [hunk 1]
```diff
@@ -15,6 +15,7 @@ namespace Dynamo.Graph.Workspaces
     /// </summary>
     public class WorkspaceInfo
     {
+        private const string dynamo1HomeWorkspaceNameString = "Home";
```
Comment (ramramps) [inline src/DynamoCore/Graph/Workspaces/WorkspaceInfo.cs]: so irrespective of languages, "Home" is always a "Home" - can you write down a small comment here? so we don't repeat the same for 3.0?

Comment (mjkkirschner) [reply src/DynamoCore/Graph/Workspaces/WorkspaceInfo.cs]: sure will do.
in I called this Dynamo1Home... as in Dynamo 2 the name of homeWorkspaces can now to be set to other values, like the filePath... but it's not an issue as in json we have the customNode flag.

File: src/DynamoCore/Graph/Workspaces/WorkspaceInfo.cs [hunk 2]
```diff
@@ -185,7 +186,8 @@ public WorkspaceInfo()
                 // we have a dyf and it lacks an ID field, we need to assign it
                 // a deterministic guid based on its name.  By doing it deterministically,
                 // files remain compatible
-                if (string.IsNullOrEmpty(id) && !string.IsNullOrEmpty(funName) && funName != Properties.Resources.DefaultHomeWorkspaceName)
+                //TODO(mjk) we should get rid of this and throw instead since non hame names are now valid in json format.
```
Comment (ramramps) [inline src/DynamoCore/Graph/Workspaces/WorkspaceInfo.cs]: home *

File: src/DynamoCore/Graph/Workspaces/WorkspaceInfo.cs [hunk 3]
```diff
@@ -184,7 +186,8 @@ public WorkspaceInfo()
...
+                if (string.IsNullOrEmpty(id) && !string.IsNullOrEmpty(funName) && funName != dynamo1HomeWorkspaceNameString)
```
Comment (mjkkirschner) [inline src/DynamoCore/Graph/Workspaces/WorkspaceInfo.cs]: @QilongTang do you remember why you added this? I guess at the time you added it here for JSON deserialization it was still valid, but now that json home workspace names can be other strings besides "Home" I think we should use the cusotmNode flag here instead or just throw if the UUID is null.

**Comment by mjkkirschner** [comment]
@QilongTang @ramramps it seems unlikely that we'll ever hit this second issue I found because json should always have the UUID - but what we do when we hit that looks wrong to me. Can be fixed later though I think.

Thoughts?

**Comment by ramramps** [comment]
we can fix it later. that's my thought

**Comment by mjkkirschner** [comment]
@ramramps I will file a task, and run the self serve CI now then.

**Comment by mjkkirschner** [comment]
@QilongTang I wanted to get a daily build with this change present as testing will be much easier then anyone trying to test with a local build because of the localization files issue. But if you would like to have more time to review, please feel free to revert this.

**Pull request merged**